Objective-C/CBLDatabase.mm

```diff
@@ -1041,11 +1041,22 @@ - (void) scheduleDocumentExpiration: (NSTimeInterval)minimumDelay {
 
 - (void) purgeExpiredDocuments {
     dispatch_async(dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        CBLLog(Database, @"Purging expired documents...");
-        C4Error err;
-        UInt64 nPurged = c4db_purgeExpiredDocs(_c4db, &err);
-        CBLLog(Database, @"Purged %lld expired documents", nPurged);
-        [self scheduleDocumentExpiration: 1.0];
+        CBL_LOCK(self) {
```

Comment:
we are always performing the purgeExpiration on Main thread, so is there a chance it is going to be thread unsafe?

Reply:
We call -purgeExpiredDocuments on the main queue but the actual purge is done inside the global queue (see line 1043) in the background not the main queue.